### PR TITLE
Formatter: always specify global `runner.dialect`.

### PR DESCRIPTION
**Pull request by dos65**
In Scalafmt `3.1.0` this setting becomes required.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala [hunk 1]
```diff
@@ -289,18 +293,40 @@ final class FormattingProvider(
...
+        item.toRelative(workspace)
+      }
+
+      val fileOverrideDirs = itemsRequiresUpgrade
+        .filter { case (_, dialect) =>
+          dialect != minDialect
+        }
+        .map { case (item, _) =>
+          item.toRelative(workspace)
+        }
```
Comment (ckipp01) [inline metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala:328]:
```suggestion
        .collect {
          case (item, dialect) if dialect != minDialect =>
            item.toRelative(workspace)
        }
```
Nit.

**Review by ckipp01 — APPROVED** [review]
Just a nit, but take it or leave it. Other than that, this looks good! LGTM!

File: metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala (hunk 1 above)
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala]: Haha. As usual. Thx!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**